worker-document/src/main/java/com/hpe/caf/worker/document/scripting/specs/AbstractScriptSpec.java

```diff
@@ -94,6 +107,13 @@ public static AbstractScriptSpec create(final ApplicationImpl application, final
      */
     public abstract boolean isStatic();
 
+    /**
+     * Returns the JavaScript engine to use for the script represented by this specification.
+     *
+     * @return the JavaScript engine to use for the script represented by this specification.
+     */
+    public abstract ScriptEngineType getEngineType();
```

Comment:
Mark this up as `@Nonnull` since the caller can expect it to never return `null`.  That is done for all functions that return non-primitives in this module so also review any other new ones that you've added.